### DISCUSSION TITLE
Skip a set amount of rows

### DISCUSSION THREAD

**Discussion by freshrebel** [JoshClose/CsvHelper#2120] [Q&A] [answered]
Hi

I would like to import a CSV but skip the first 12 rows (as these are incompatible with my classmap).
What is the best way to do this?

Right now I have this:
```
using (var reader = new StreamReader(txtFilePath.Text))
using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
{
  csv.Context.RegisterClassMap<TransactionClassMap>();
  var transactions = csv.GetRecords<Transaction>().Skip(12).ToList();
  FinancialData.Transactions = transactions;
}
```
But this tries to convert the lines to a Transaction and then tries to skip the first 12 rows, but I get errors in the converting so I want to skip before the conversion.

Hopefully this is possible.
Thanks in advance.

Kind regards
Arno

**Comment by JoshClose** [comment, marked as answer]
For the `Skip` call to work, the entire data set is read into memory first. There is no LINQ provider.

You can skip like this though.

```cs
for (var i = 0; i < 12; i++)
{
    csv.Read();
}
```

If you have a header and it doesn't start on line 13, you'll need to read that separately.

This also reads CSV rows, so if there are double quotes and other things in there you would want to skip file rows.

```cs
for (var i = 0; i < 12; i++)
{
    reader.ReadLine();
}
```

**Reply by freshrebel** [reply]
Tnx bro

**Reply by kris3713** [reply]
@JoshClose I know this issue is old, but I just want to say thank you implementing a great CSV parser when Microsoft themselves could not.

This discussion really helped solved the issue of wanting to skip the first row using CSVHelper.
